lib/src/events.dart

```diff
@@ -13,8 +13,14 @@ class Emitter {
 
   Stream<Event> operator[] (type) => _evtController.stream.where((e) => e.type == type);
 
+  Stream<T> getEventStream<T extends Event>(String type) {
+    final controller = new StreamController<T>();
+    _evtController.stream.where((e) => e.type == type).listen(controller.add, onDone: controller.close);
+    return controller.stream;
```

Comment:
Is this controller necessary, or could it just be the following?
```dart
return _evtController.stream.where((e) => e.type == type);
```

Also, I believe with the current implementation, any `onError` handler added to the returned stream would have no effect, and errors emitted by the original stream would be uncaught.

Reply:
I tried using `Stream.where()`, but that doesn't give you a properly typed Stream (I believe we'd need the `whereType` Dart 2 method to do that).

Reply:
added the onError handler